tests/auth_oidc_refresh_test.go

```diff
@@ -96,6 +172,31 @@ func Test_Authenticate_OIDC_Refresh(t *testing.T) {
 			ctx.NotNil(newSession)
 			ctx.NotNil(newSession.GetToken())
 			ctx.NotEqual(newSession.GetToken(), apiSession.GetToken())
+			ctx.IsType(newSession, &edge_apis.ApiSessionOidc{})
+
+			newTokens := newSession.(*edge_apis.ApiSessionOidc)
+
+			t.Run("the new token has the correct audiences", func(t *testing.T) {
+				ctx.testContextChanged(t)
+
+				newAccessClaims := &jwt.MapClaims{}
+				newAccessToken, _, err := jwtParser.ParseUnverified(newTokens.OidcTokens.AccessToken, newAccessClaims)
+				ctx.Req.NoError(err)
+				ctx.Req.NotNil(newAccessToken)
+				newAccessAudiences, err := newAccessToken.Claims.GetAudience()
+				ctx.Req.NoError(err)
+				ctx.Req.Len(newAccessAudiences, 1, "expected 1 audience")
+				ctx.Equal(originalAccessAudiences[0], newAccessAudiences[0])
+
+				newRefreshClaims := &jwt.MapClaims{}
+				newRefreshToken, _, err := jwtParser.ParseUnverified(newTokens.OidcTokens.RefreshToken, newRefreshClaims)
+				ctx.Req.NoError(err)
+				ctx.Req.NotNil(newRefreshToken)
+				newRefreshAudiences, err := newRefreshToken.Claims.GetAudience()
+				ctx.Req.NoError(err)
+				ctx.Req.Len(newRefreshAudiences, 1, "expected 1 audience")
+				ctx.Equal(originalRefreshAudiences[0], newRefreshAudiences[0])
+			})
```

Comment:
no test for ClaimLegacyNative in the tests. just making sure that's on purpose?

Reply:
i don't see any negative tests right?

Reply:
Because it is going to be removed here shortly.